pkg/operator/reports_test.go

```diff
@@ -92,3 +102,235 @@ func TestGetNextReportPeriod(t *testing.T) {
+			name: "when status running is false and reason is Scheduled return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ScheduledReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is true and reason is Scheduled return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionTrue, util.ScheduledReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is false and reason is InvalidReport return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.InvalidReportReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is true and reason is InvalidReport return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionTrue, util.InvalidReportReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is false and reason is RunImmediately return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.RunImmediatelyReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is true and reason is RunImmediately return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionTrue, util.RunImmediatelyReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+	}
+
+	for _, testCase := range testTable {
+		var mockLogger = logrus.New()
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			runningCond := isReportFinished(mockLogger, testCase.report)
+			assert.Equalf(t, runningCond, testCase.expectFinished, "expected the report '%s' would return '%t', but got '%t'", testCase.name, testCase.expectFinished, runningCond)
```

Comment:
This message is repetitive (and the report name isn't the test case name).

Reply:
Updated.